### ISSUE TITLE
Add a post-process step to normalize the data?

### ISSUE THREAD

**Issue by ashi009** [NSHipster/XcodeBuildSettings.com#26]
This project has been a great help in terms of understanding Xcode build settings. Thank you very much for the work.

When processing the generated JSON file, there are some issues with the data itself. The most difficult part is to figure out the schema of the file. 

For instance, the `type` field consists of 19 different values, only 13 of them are unique by definition. The type of `values` field is also unclear, some of them are `string[]`, while the rest are `Array<{ Value: string,  ​DisplayName?: string, ​CommandLineArgs: string[]}>`.

I'm wondering if it's a good idea to add a post-process step to normalize all the setting descriptors while providing a JSON schema.

**Comment by mattt** [comment]
I agree that the data files are confusing and inconsistent. However, this is perhaps expected for configuration introduced in different parts of a large software project (the Xcode build system) over the course of many years.

Introducing a post-processing step and/or a schema could provide some benefit to end-users. If you or someone else wanted to develop this feature, I'd be happy to review it. However, I don't see a pressing need to implement this at the moment.

**Comment by ashi009** [comment]
> I agree that the data files are confusing and inconsistent. However, this is perhaps expected for configuration introduced in different parts of a large software project (the Xcode build system) over the course of many years.
> 
> 
> 
> Introducing a post-processing step and/or a schema could provide some benefit to end-users. If you or someone else wanted to develop this feature, I'd be happy to review it. However, I don't see a pressing need to implement this at the moment.

Good to know you are open to this. We already have something working internally, will send you a PR once the legal process approves.
